Mathlib/GroupTheory/Exponent.lean

```diff
@@ -374,6 +389,23 @@ end CancelCommMonoid
 
 end Monoid
 
+section Group
+
+@[to_additive AddGroup.one_lt_exponent]
```

Comment:
Do you really need `Group` here?

Reply:
Yes. The issue is that you need cancellation, for otherwise the exponent could be zero. And as pointed out above, a finite (left- or right-) cancellative monoid is a group. Just to drive home the point, any nontrivial `MonoidWithZero` has exponent zero.
```lean
import Mathlib.GroupTheory.Exponent

example {M : Type*} [MonoidWithZero M] [Nontrivial M] :
    Monoid.exponent M = 0 := by
  rw [@Monoid.exponent_eq_zero_iff, Monoid.ExponentExists]
  push_neg
  exact fun n hn ↦ ⟨0, by simp [zero_pow hn]⟩
```